### PR TITLE
Replace some instances of eprintf with R_LOG_

### PR DESCRIPTION
**Pull request by terorie**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -1720,7 +1720,7 @@ R_API bool r_anal_function_add_bb(RAnal *a, RAnalFunction *fcn, ut64 addr, ut64
 	}
 
 	if (!block) {
-		D eprintf ("Warning: r_anal_function_add_bb failed in fcn 0x%08"PFMT64x" at 0x%08"PFMT64x"\n", fcn->addr, addr);
+		D R_LOG_WARN ("Warning: r_anal_function_add_bb failed in fcn 0x%08"PFMT64x" at 0x%08"PFMT64x"", fcn->addr, addr);
```
Comment (terorie) [inline libr/anal/fcn.c:1723]:
```suggestion
		D R_LOG_WARN ("Warning: r_anal_function_add_bb failed in fcn 0x%08"PFMT64x" at 0x%08"PFMT64x, fcn->addr, addr);
```

File: libr/anal/fcn.c [hunk 2]
```diff
@@ -1775,7 +1775,7 @@ R_API int r_anal_function_complexity(RAnalFunction *fcn) {
 	r_list_foreach (fcn->bbs, iter, bb) {
 		N++; // nodes
 		if ((!anal || anal->verbose) && bb->jump == UT64_MAX && bb->fail != UT64_MAX) {
-			eprintf ("Warning: invalid bb jump/fail pair at 0x%08"PFMT64x" (fcn 0x%08"PFMT64x"\n", bb->addr, fcn->addr);
+			R_LOG_WARN ("Warning: invalid bb jump/fail pair at 0x%08"PFMT64x" (fcn 0x%08"PFMT64x"", bb->addr, fcn->addr);
```
Comment (terorie) [inline libr/anal/fcn.c:1778]:
```suggestion
			R_LOG_WARN ("Warning: invalid bb jump/fail pair at 0x%08"PFMT64x" (fcn 0x%08"PFMT64x, bb->addr, fcn->addr);
```

**Review by terorie — COMMENTED** [review]
some bugs

File: libr/core/blaze.c [hunk 3]
```diff
@@ -38,7 +38,7 @@ static bool __is_data_block_cb(RAnalBlock *block, void *user) {
 
 static int __isdata(RCore *core, ut64 addr) {
 	if (!r_io_is_valid_offset (core->io, addr, false)) {
-		// eprintf ("Warning: Invalid memory address at 0x%08"PFMT64x"\n", addr);
+		// R_LOG_WARN ("Warning: Invalid memory address at 0x%08"PFMT64x"", addr);
```
Comment (terorie) [inline libr/core/blaze.c:41]:
```suggestion
		// R_LOG_WARN ("Warning: Invalid memory address at 0x%08"PFMT64x, addr);
```

File: libr/core/canal.c [hunk 4]
```diff
@@ -5994,12 +5994,12 @@ R_API void r_core_anal_inflags(RCore *core, const char *glob) {
 			continue;
 		}
 		if (a0 > a1) {
-			eprintf ("Warning: unsorted flag list 0x%"PFMT64x" 0x%"PFMT64x"\n", a0, a1);
+			R_LOG_WARN ("Warning: unsorted flag list 0x%"PFMT64x" 0x%"PFMT64x"", a0, a1);
```
Comment (terorie) [inline libr/core/canal.c:5997]:
```suggestion
			R_LOG_WARN ("Warning: unsorted flag list 0x%"PFMT64x" 0x%"PFMT64x, a0, a1);
```

File: libr/core/canal.c [hunk 5]
```diff
@@ -5994,12 +5994,12 @@ R_API void r_core_anal_inflags(RCore *core, const char *glob) {
...
 			continue;
 		}
 		st64 sz = a1 - a0;
 		if (sz < 1 || sz > core->anal->opt.bb_max_size) {
-			eprintf ("Warning: invalid flag range from 0x%08"PFMT64x" to 0x%08"PFMT64x"\n", a0, a1);
+			R_LOG_WARN ("Warning: invalid flag range from 0x%08"PFMT64x" to 0x%08"PFMT64x"", a0, a1);
```
Comment (terorie) [inline libr/core/canal.c:6002]:
```suggestion
			R_LOG_WARN ("Warning: invalid flag range from 0x%08"PFMT64x" to 0x%08"PFMT64x, a0, a1);
```

File: libr/debug/p/native/w32.c [hunk 6]
```diff
@@ -662,8 +662,8 @@ static void printwincontext(HANDLE hThread, CONTEXT *ctx) {
 #if _WIN64
 	eprintf ("ControlWord   = %08x StatusWord   = %08x\n", ctx->FltSave.ControlWord, ctx->FltSave.StatusWord);
 	eprintf ("MxCsr         = %08x TagWord      = %08x\n", ctx->MxCsr, ctx->FltSave.TagWord);
-	eprintf ("ErrorOffset   = %08x DataOffset   = %08x\n", ctx->FltSave.ErrorOffset, ctx->FltSave.DataOffset);
-	eprintf ("ErrorSelector = %08x DataSelector = %08x\n", ctx->FltSave.ErrorSelector, ctx->FltSave.DataSelector);
+	R_LOG_ERROR ("ErrorOffset   = %08x DataOffset   = %08x", ctx->FltSave.ErrorOffset, ctx->FltSave.DataOffset);
+	R_LOG_ERROR ("ErrorSelector = %08x DataSelector = %08x", ctx->FltSave.ErrorSelector, ctx->FltSave.DataSelector);
```
Comment (terorie) [inline libr/debug/p/native/w32.c:666]: revert

File: libr/debug/p/native/w32.c [hunk 7]
```diff
@@ -685,8 +685,8 @@ static void printwincontext(HANDLE hThread, CONTEXT *ctx) {
 #else
 	eprintf ("ControlWord   = %08x StatusWord   = %08x\n", (ut32) ctx->FloatSave.ControlWord, (ut32) ctx->FloatSave.StatusWord);
 	eprintf ("MxCsr         = %08x TagWord      = %08x\n", *(ut32 *)&ctx->ExtendedRegisters[24], (ut32)ctx->FloatSave.TagWord);
-	eprintf ("ErrorOffset   = %08x DataOffset   = %08x\n", (ut32)ctx->FloatSave.ErrorOffset, (ut32)ctx->FloatSave.DataOffset);
-	eprintf ("ErrorSelector = %08x DataSelector = %08x\n", (ut32)ctx->FloatSave.ErrorSelector, (ut32) ctx->FloatSave.DataSelector);
+	R_LOG_ERROR ("ErrorOffset   = %08x DataOffset   = %08x", (ut32)ctx->FloatSave.ErrorOffset, (ut32)ctx->FloatSave.DataOffset);
+	R_LOG_ERROR ("ErrorSelector = %08x DataSelector = %08x", (ut32)ctx->FloatSave.ErrorSelector, (ut32) ctx->FloatSave.DataSelector);
```
Comment (terorie) [inline libr/debug/p/native/w32.c:689]: revert

**Review by trufae — CHANGES_REQUESTED** [review]
Cmdline tools probably are fine to report errors via eprintf (see failing tests)

File: libr/debug/p/native/windows/windows_debug.c [hunk 8]
```diff
@@ -220,8 +220,8 @@ static void __printwincontext(HANDLE th, CONTEXT *ctx) {
 #if _WIN64
 	eprintf ("ControlWord   = %08x StatusWord   = %08x\n", ctx->FltSave.ControlWord, ctx->FltSave.StatusWord);
 	eprintf ("MxCsr         = %08lx TagWord      = %08x\n", ctx->MxCsr, ctx->FltSave.TagWord);
-	eprintf ("ErrorOffset   = %08lx DataOffset   = %08lx\n", ctx->FltSave.ErrorOffset, ctx->FltSave.DataOffset);
-	eprintf ("ErrorSelector = %08x DataSelector = %08x\n", ctx->FltSave.ErrorSelector, ctx->FltSave.DataSelector);
+	R_LOG_ERROR ("ErrorOffset   = %08lx DataOffset   = %08lx", ctx->FltSave.ErrorOffset, ctx->FltSave.DataOffset);
+	R_LOG_ERROR ("ErrorSelector = %08x DataSelector = %08x", ctx->FltSave.ErrorSelector, ctx->FltSave.DataSelector);
```
Comment (terorie) [inline libr/debug/p/native/windows/windows_debug.c:224]: revert

File: libr/debug/p/native/windows/windows_message.c [hunk 9]
```diff
@@ -316,7 +316,7 @@ R_API void r_w32_identify_window(void) {
 		return;
 	}
 	if (!win) {
-		eprintf ("Error trying to get information from 0x%08"PFMT64x"\n", (ut64)hwnd);
+		R_LOG_ERROR ("Error trying to get information from 0x%08"PFMT64x"", (ut64)hwnd);
```
Comment (terorie) [inline libr/debug/p/native/windows/windows_message.c:319]:
```suggestion
		R_LOG_ERROR ("Error trying to get information from 0x%08"PFMT64x, (ut64)hwnd);
```

**Review by terorie — COMMENTED** [review]
some bugs

File: sys/lint.sh [hunk 10]
```diff
@@ -4,6 +4,6 @@
 (git grep 'for (long' | grep -v sys/) && exit 1
 (git grep 'for (size_t' | grep -v sys/) && exit 1
 
-(git grep 'R_LOG_' | grep '\\n' | grep -v sys/) ## && exit 1
+(git grep 'R_LOG_' | grep '\\n' | grep -v sys/) && exit 1
```
Comment (terorie) [inline sys/lint.sh:7]:
```suggestion
(git grep 'R_LOG_' | grep '\\n' | grep -v sys/) ## && exit 1
```

File: libr/util/str.c [hunk 11]
```diff
@@ -1200,7 +1200,7 @@ R_API int r_str_unescape(char *buf) {
 			err |= r_hex_to_byte (&ch,  buf[i + 2]);
 			err |= r_hex_to_byte (&ch2, buf[i + 3]);
 			if (err) {
-				eprintf ("Error: Non-hexadecimal chars in input.\n");
+				R_LOG_ERROR ("Error: Non-hexadecimal chars in input.");
```
Comment (trufae) [inline libr/util/str.c:1203]: Same

File: sys/lint.sh [hunk 12]
```diff
@@ -4,6 +4,6 @@
 (git grep 'for (long' | grep -v sys/) && exit 1
 (git grep 'for (size_t' | grep -v sys/) && exit 1
 
-(git grep 'R_LOG_' | grep '\\n' | grep -v sys/) ## && exit 1
+(git grep 'R_LOG_' | grep '\\n' | grep -v sys/) ## && exit 1
```
Comment (trufae) [inline sys/lint.sh:7]:
```suggestion
(git grep 'R_LOG_' | grep '\\n' | grep -v sys/) && exit 1
```

File: libr/util/str.c (hunk 11 above)
Comment (terorie) [reply libr/util/str.c]: fixed all instances

File: libr/anal/dwarf_process.c [hunk 13]
```diff
@@ -273,7 +273,7 @@ static st32 parse_type(Context *ctx, const ut64 offset, RStrBuf *strbuf, ut64 *s
 		*visited = su;
 	}
 	if (visited && set_u_contains (*visited, offset)) {
-		eprintf ("Warning: anal.dwarf.parse_type: infinite recursion detected.\n");
+		R_LOG_WARN ("Warning: anal.dwarf.parse_type: infinite recursion detected.");
```
[the remark this answers is not in the document]
Comment (terorie) [reply libr/anal/dwarf_process.c]: fixed all instances.

File: libr/core/canal.c [hunk 14]
```diff
@@ -759,7 +759,7 @@ static bool __core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int de
 	const char *cc = r_anal_cc_default (core->anal);
 	if (!cc) {
 		if (r_anal_cc_once (core->anal)) {
-			eprintf ("Warning: set your favourite calling convention in `e anal.cc=?`\n");
+			R_LOG_WARN ("set your favourite calling convention in `e anal.cc=?`");
```
Comment (trufae) [inline libr/core/canal.c:762]: revert this line as it seems to break some tests. we will look later at them
```suggestion
			eprintf ("Warning: set your favourite calling convention in `e anal.cc=?`\n");
```

File: libr/asm/asm.c [hunk 15]
```diff
@@ -1125,7 +1125,7 @@ R_API RAsmCode *r_asm_massemble(RAsm *a, const char *assembly) {
 			}
 			if (stage == STAGES - 1) {
 				if (ret < 1) {
-					eprintf ("Cannot assemble '%s' at line %d\n", ptr_start, linenum);
+					R_LOG_ERROR ("Cannot assemble '%s' at line %d", ptr_start, linenum);
```
Comment (trufae) [inline libr/asm/asm.c:1128]:
```suggestion
					eprintf ("Cannot assemble '%s' at line %d\n", ptr_start, linenum);
```

File: libr/bin/p/bin_elf.inc [hunk 16]
```diff
@@ -984,7 +984,7 @@ static RList* patch_relocs(RBin *b) {
 		return NULL;
 	}
 	if (!io->cached) {
-		eprintf ("Warning: run r2 with -e bin.cache=true to fix relocations in disassembly\n");
+		R_LOG_WARN ("run r2 with -e bin.cache=true to fix relocations in disassembly");
```
Comment (trufae) [inline libr/bin/p/bin_elf.inc:987]:
```suggestion
		eprintf ("Warning: run r2 with -e bin.cache=true to fix relocations in disassembly\n");
```

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/anal/fcn.c [hunk 17]
```diff
@@ -1690,12 +1690,12 @@ R_API RAnalFunction *r_anal_get_function_byname(RAnal *a, const char *name) {
...
 	D eprintf ("Add bb\n");
 	if (size == 0) { // empty basic blocks allowed?
-		eprintf ("Warning: empty basic block at 0x%08"PFMT64x" is not allowed. pending discussion.\n", addr);
+		R_LOG_WARN ("empty basic block at 0x%08"PFMT64x" is not allowed. pending discussion.", addr);
 		r_warn_if_reached ();
 		return false;
 	}
 	if (size > a->opt.bb_max_size) {
-		eprintf ("Warning: can't allocate such big bb of %"PFMT64d" bytes at 0x%08"PFMT64x"\n", (st64)size, addr);
+		R_LOG_WARN ("can't allocate such big bb of %"PFMT64d" bytes at 0x%08"PFMT64x"", (st64)size, addr);
```
Comment (trufae) [inline libr/anal/fcn.c:1698]:
```suggestion
		R_LOG_WARN ("can't allocate such big bb of %"PFMT64d" bytes at 0x%08"PFMT64x, (st64)size, addr);
```

File: libr/anal/rtti_msvc.c [hunk 18]
```diff
@@ -896,7 +896,7 @@ static const char *recovery_apply_complete_object_locator(RRTTIMSVCAnalContext *
 
 	if (!col->td) {
 		if (context->vt_context->anal->verbose) {
-			eprintf ("Warning: no td for col at 0x%"PFMT64x"\n", col->addr);
+			R_LOG_WARN ("no td for col at 0x%"PFMT64x"", col->addr);
```
Comment (trufae) [inline libr/anal/rtti_msvc.c:899]:
```suggestion
			R_LOG_WARN ("no td for col at 0x%"PFMT64x, col->addr);
```

**Comment by trufae** [comment]
Thanks!

**Pull request merged**